### ISSUE TITLE
coursier pull incorrect dependency

### ISSUE THREAD

**Issue by xuwei-k** [sbt/sbt#6748]
## steps

JDK 8

#### `build.sbt`

```scala
libraryDependencies += "com.zaxxer" % "HikariCP" % "5.0.0"
```

#### `project/build.properties`

```
sbt.version=1.5.6
```

run `sbt "print externalDependencyClasspath" "set useCoursier := false" "print externalDependencyClasspath"`

#### default (enable coursier) 

```
* Attributed(/Users/kenji/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/zaxxer/HikariCP/5.0.0/HikariCP-5.0.0.jar)
* Attributed(/Users/kenji/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/slf4j/slf4j-api/2.0.0-alpha1/slf4j-api-2.0.0-alpha1.jar)
```

#### disable coursier with sbt 1.5.6 (or sbt 1.2.8)

```
* Attributed(/Users/kenji/.ivy2/cache/com.zaxxer/HikariCP/bundles/HikariCP-5.0.0.jar)
* Attributed(/Users/kenji/.ivy2/cache/org.slf4j/slf4j-api/jars/slf4j-api-1.7.30.jar)
```

## expectation

I think `1.7.30` is correct version. not `2.0.0-alpha1`.

## notes

https://repo1.maven.org/maven2/com/zaxxer/HikariCP/5.0.0/HikariCP-5.0.0.pom

Here is simplified pom.xml.

Maybe coursier use `project.profiles.profile.properties` instead of `project.properties` 🤔 ❓ 

```scala
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>

   <properties>
      <slf4j.version>1.7.30</slf4j.version>
   </properties>

   <groupId>com.zaxxer</groupId>
   <artifactId>HikariCP</artifactId>
   <version>5.0.0</version>
   <packaging>bundle</packaging>

   <name>HikariCP</name>
   <description>Ultimate JDBC Connection Pool</description>
   <url>https://github.com/brettwooldridge/HikariCP</url>

   <organization>
      <name>Zaxxer.com</name>
      <url>https://github.com/brettwooldridge</url>
   </organization>

   <dependencies>
      <dependency>
         <groupId>org.slf4j</groupId>
         <artifactId>slf4j-api</artifactId>
         <version>${slf4j.version}</version>
      </dependency>
   </dependencies>

   <profiles>
      <profile>
         <id>Java11</id>
         <activation>
            <jdk>[11,)</jdk>
         </activation>
         <properties>
            <!-- sureFireOptions11>
               -add-opens java.base/java.net=ALL-UNNAMED
               -add-opens java.base/java.security=ALL-UNNAMED
               -add-exports java.base/sun.net.www.protocol.http=ALL-UNNAMED
               -add-exports java.base/sun.net.www.protocol.https=ALL-UNNAMED
            </sureFireOptions11 -->
            <slf4j.version>2.0.0-alpha1</slf4j.version>
            <sureFireForks11>true</sureFireForks11>
         </properties>
         <dependencies>
            <dependency>
               <groupId>org.apache.logging.log4j</groupId>
               <artifactId>log4j-slf4j18-impl</artifactId>
               <version>${log4j.version}</version>
               <scope>test</scope>
            </dependency>
         </dependencies>
         <build>
            <plugins>
               <plugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-compiler-plugin</artifactId>
                  <version>3.8.1</version>
                  <executions>
                     <execution>
                        <id>compile-java11</id>
                        <goals>
                           <goal>compile</goal>
                        </goals>
                        <configuration>
                           <release>11</release>
                           <compileSourceRoots>
                              <compileSourceRoot>${project.basedir}/src/main/java11</compileSourceRoot>
                           </compileSourceRoots>
                           <multiReleaseOutput>true</multiReleaseOutput>
                        </configuration>
                     </execution>
                  </executions>
               </plugin>
            </plugins>
         </build>
      </profile>

   </profiles>
</project>
```

/cc @alexarchambault

**Comment by eed3si9n** [comment]
@xuwei-k Thanks for the report. Could you also report this to coursier/coursier please?

**Comment by eed3si9n** [comment]
Actually this might be the correct behavior if you're building on JDK 11? https://maven.apache.org/guides/introduction/introduction-to-profiles.html#details-on-profile-activation

```xml
<profile>
  <id>Java11</id>
  <activation>
    <jdk>[11,)</jdk>
  </activation>
....
```

**Comment by alexarchambault** [comment]
I didn't try the sbt repro here, but oddly enough, I can't reproduce that on the command-line of coursier:
```text
$ cs launch coursier:2.0.15 --jvm 8 --scala 2.12.15 -- resolve com.zaxxer:HikariCP:5.0.0
com.zaxxer:HikariCP:5.0.0:default
org.slf4j:slf4j-api:1.7.30:default

$ cs launch coursier:2.0.15 --jvm 11 --scala 2.12.15 -- resolve com.zaxxer:HikariCP:5.0.0
com.zaxxer:HikariCP:5.0.0:default
org.slf4j:slf4j-api:2.0.0-alpha1:default
```

(sbt 1.5.6 uses coursier `2.0.15`)

**Comment by eed3si9n** [comment]
Circling back to this, I think Yoshida-san was saying the fact that it resolves to `slf4j-api:2.0.0-alpha1` as a problem and expected:

> I think `1.7.30` is correct version. not `2.0.0-alpha1`.

so I think both Coursier CLI and sbt reproduces 2.0.0-alpha1, but I think Alex and I are on the same page that this works as designed per POM specification.

**Comment by henricook** [comment]
@eed3si9n Thanks as always for your great work here. I've just dealt with this by pinning slf4j-api on my project which means we can finally use Coursier with SBT, about 3 years after it became the default 😂 

To be clear here so I understand, we're saying the slf4j project has done something funky with their POM that makes this version get chosen, and as a result it's their problem and not Coursier's?
